pkg/common/util/signals.go

```diff
@@ -2,20 +2,17 @@ package util
 
 import (
 	"context"
-	"os"
 	"os/signal"
 	"syscall"
 )
 
-func SignalListener(ctx context.Context, cancel func()) {
+func SignalListener(parent context.Context, cancel func()) {
 	go func() {
-		signalCh := make(chan os.Signal, 1)
-		signal.Notify(signalCh, syscall.SIGINT, syscall.SIGTERM)
+		ctx, cancel := signal.NotifyContext(parent, syscall.SIGINT, syscall.SIGTERM)
 
 		select {
+		case <-parent.Done():
 		case <-ctx.Done():
-			return
-		case <-signalCh:
 			cancel()
```

Comment:
Line 11 now declares a `cancel` variable that is going to mask the one provided in the arguments, which will prevent this function from cancelling the parent context when a signal is received. Both the cancel in the argument and the cancel for the child context need to be called in order to cancel the parent and clean up resources.

Reply:
I removed `utils.SignalListener`.